src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerListener.cs

```diff
@@ -148,5 +181,253 @@ private void InitializeBuilder()
                     .WithObserverFactory(this);
             }
         }
+
+        private async Task<IRemainingWorkEstimator> GetWorkEstimatorAsync()
+        {
+            if (_workEstimatorBuilder == null)
+            {
+                _workEstimatorBuilder = new ChangeFeedProcessorBuilder()
```

Comment:
Just wondering, since I don't know the nature of the code that calls this, there won't be a change of it being called concurrently, right? Otherwise this might need a semaphore or lock